### PR TITLE
Boost: Fix first user flow issues

### PR DESCRIPTION
**Pull request by haqadn**
Fixes Automattic/boost-cloud#157

#### Changes proposed in this Pull Request:
* Constrained the width of the container in boost dashboard and pricing page
* Added promo text to indicate that the discount is for first year only

#### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
pc9hqz-1fV-p2#comment-943

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Open the admin.php?page=jetpack-boost#/getting-started.
* Make sure the table constrains to 1320px on very large monitors.
* Make sure it has the badge "First Year Discount" under the price for premium plan.

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/react-components/BoostPricingTable.tsx
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/react-components/BoostPricingTable.tsx]: $AUD doesn't discount exactly 50%. I don't think we can use the exact discount amount as a gate here.

However, if you look at `Wpcom_Products::get_product_pricing`, `discount_price` only differs from `full_price` if there is introductory pricing. It shows how to read the `introductory_pricing`.

IMHO we can't guarantee that `Wpcom_Products::get_product_pricing` will only ever return `introductory` pricing - we may have other reasons for discount in the future.

So I think the "correct" fix is to add a method to `Wpcom_Products` which returns true if there is introductory pricing, so that the UI can show that badge. Look at `Wpcom_Products::get_product_pricing` for how to check for the presence of intro pricing.

It means you'll need a review from someone with approver rights over My-Jetpack, but I'm pretty sure @oskosk can help with that ;)

**Review by thingalon — CHANGES_REQUESTED** [review]

**Review by thingalon — DISMISSED** [review]
This looks good to me! Though it will need a 👍 from someone with approver access to My Jetpack (@oskosk ?)

**Review by thingalon — DISMISSED** [review]

**Review by oskosk — DISMISSED** [review]
LGTM

**Review by kraftbj — APPROVED** [review]

**Review by oskosk — APPROVED** [review]

**Comment by haqadn** [comment]
Cherry-Picked 31d1f60eb34e

**Pull request merged**
